src/pgwire/src/protocol.rs

```diff
@@ -360,15 +360,29 @@ where
                 let execute_root_span =
                     tracing::debug_span!(parent: None, "advance_ready", otel.name = message_name);
                 execute_root_span.follows_from(tracing::Span::current());
-                self.execute(
-                    portal_name,
-                    max_rows,
-                    portal_exec_message,
-                    None,
-                    ExecuteTimeout::None,
-                )
-                .instrument(execute_root_span)
-                .await?
+                let state = self
+                    .execute(
```

Comment:
The pgwire execute message can specify `max_rows` which causes a portal to partially consume the result set, and the client can ask for more data later. `portals.pt` didn't change, but that seems to be because it always issues a `BEGIN` during its testing, which I suppose is ok? I think this means that portals.pt could have tested without using BEGIN, but happens to not do that, and now we can't do that. Maybe there was a reason I didn't add any tests without BEGIN there, but I can't think of it now.

Actually, one of the tests there is now misleading: https://github.com/MaterializeInc/materialize/blob/f06db04739f78ce3eba0db77f8ec493be8103d5a/test/pgtest/portals.pt#L76 is testing that portals are destroyed after Sync, but we're actually destroying it after Execute. Can you add something like:

```
send
Parse {"query": "VALUES (1), (2)"}
Bind {"portal": "c"}
Execute {"portal": "c", "max_rows": 1}
Execute {"portal": "c", "max_rows": 1}
Sync
----

until
ReadyForQuery
----
```

to a new file at `test/pgtest-mz/portals.pt` with a comment describing that we differ from postgres?

Reply:
I'm getting the following in Materialize

```
send
Parse {"query": "VALUES (1), (2)"}
Bind {"portal": "c"}
Execute {"portal": "c", "max_rows": 1}
Execute {"portal": "c", "max_rows": 1}
Sync
----

until
ReadyForQuery
----
ParseComplete
BindComplete
DataRow {"fields":["1"]}
PortalSuspended
ErrorResponse {"fields":[{"typ":"S","value":"ERROR"},{"typ":"C","value":"34000"},{"typ":"M","value":"portal \"c\" does not exist"}]}
ReadyForQuery {"status":"I"}
```

I'm not really sure how to interpret that, but I'll look into what's going on. Maybe we remove the portal after committing, which I *think* we're not supposed to do.

Reply:
Ok I moved the test with the misleading comment to `pgtest-mz` and updated the comment. Also I added the test above.

Reply:
Yeah, that response is exactly what I expected to get after this PR. It's like a tiiiiiny bit unfortunate, but such a small deal it really doesn't matter. It's another one of those "welp, add a BEGIN if you really need that" things.

Reply:
Yeah I agree.